src/streaming/MediaPlayer.js

```diff
@@ -270,6 +278,8 @@ MediaPlayer = function (context) {
                     doAutoPlay.call(this);
                 }
             }
+
+            this.adapter.reset();
```

Comment:
@KozhinM I'm fine with moving it to the top.  As long as it is called before `doAutoPlay.call()`.